<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ChatWindow</class>
 <widget class="QMainWindow" name="ChatWindow">
  <property name="geometry">
   <rect>
    <x>100</x>
    <y>100</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chat Client</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QWidget" name="contactsWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_Contacts">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <property name="contentsMargins">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>0</width>
         <height>0</height>
        </rect>
       </property>
       <item>
        <widget class="QWidget" name="contactsHeaderWidget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_ContactsHeader">
          <property name="contentsMargins">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>10</width>
            <height>10</height>
           </rect>
          </property>
          <item>
           <widget class="QLabel" name="contactsLabel">
            <property name="text">
             <string>Contacts</string>
            </property>
            <property name="font">
             <font>
              <pointsize>14</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_ContactsHeader">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QListView" name="contactsListView">
         <property name="styleSheet">
          <string notr="true">QListView::item {
    padding: 8px;
    border-bottom: 1px solid #eeeeee;
}

QListView::item:selected {
    background-color: #e3f2fd;
    color: #0d47a1;
}</string>
         </property>
         <property name="uniformItemSizes">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="chatWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_Chat">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <property name="contentsMargins">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>0</width>
         <height>0</height>
        </rect>
       </property>
       <item>
        <widget class="QWidget" name="chatHeaderWidget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_ChatHeader">
          <property name="contentsMargins">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>10</width>
            <height>10</height>
           </rect>
          </property>
          <item>
           <widget class="QLabel" name="contactNameLabel">
            <property name="text">
             <string>Select a contact</string>
            </property>
            <property name="font">
             <font>
              <pointsize>14</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="contactStatusLabel">
            <property name="text">
             <string/>
            </property>
            <property name="styleSheet">
             <string notr="true">color: #666666;
font-size: 12px;</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_ChatHeader">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="chatTextEdit">
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Select a contact to start chatting&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="messageInputWidget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_MessageInput">
          <property name="contentsMargins">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>10</width>
            <height>10</height>
           </rect>
          </property>
          <item>
           <widget class="QLineEdit" name="messageLineEdit">
            <property name="placeholderText">
             <string>Type your message here...</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="sendButton">
            <property name="text">
             <string>Send</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLogout"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionLogout">
   <property name="text">
    <string>Logout</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>contactsListView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>ChatWindow</receiver>
   <slot>onContactSelected(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sendButton</sender>
   <signal>clicked()</signal>
   <receiver>ChatWindow</receiver>
   <slot>onSendButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>750</x>
     <y>550</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>messageLineEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>ChatWindow</receiver>
   <slot>onMessageReturnPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>550</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogout</sender>
   <signal>triggered()</signal>
   <receiver>ChatWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>QApplication</receiver>
   <slot>quit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
